ui/app/AppLayouts/Chat/views/ChatMessagesView.qml

```diff
@@ -160,11 +160,8 @@ Item {
 
     Loader {
         id: loadingMessagesView
-
-        readonly property bool show: !messageStore.firstUnseenMessageLoaded ||
-                                     !messageStore.initialMessagesLoaded
-        active: show
-        visible: show
+        active: !messageStore.initialMessagesLoaded
```

Comment:
That's probably wrong. We should show `loadingMessagesView` until first unseen message is loaded. Note, that each time the chat is made active it should scroll to first unseen message.

Reply:
From the initial design, the idea was to show the loading state until initialMessagesLoaded will be loaded and we'll receive the first time the unseen message id. That will mean that we have all the necessary data to display the chat and not wait for any loading data (and no freezes). After that, we do not need this loader UI as we have all the necessary loaded data in the backend.

Reply:
The first unseen message can change over time.
1) you read all messages in chat A
2) you go to chat B and then over a period of time 100 messages arrive to chat A
3) you go back to chat A, you want the view to be scrolled to first unseen message there (100 messages up)

Reply:
Thank you, I understand this, but question:
When I receive a new message to chat A and I'm on chat B - the badge notifies me that I have an unread message at chat A. If I know that chat A has an unread msg, why do I need to request the first unseen msg ID again on opening chat A?

Reply:
Yes, it is possible to optimize it and not request first unseen message if we are switching to the chat that doesn't have unread messages. Otherwise, if chat has unread messages, we should request first unseen message when making the chat active.